protocol/client/messenger_test.go

```diff
@@ -0,0 +1,62 @@
+package client
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/status-im/status-console-client/protocol/v1"
+	"github.com/stretchr/testify/require"
+)
+
+type requestsMock struct {
+	requests []protocol.RequestOptions
+}
+
+func (proto *requestsMock) Send(ctx context.Context, data []byte, options protocol.SendOptions) ([]byte, error) {
+	return nil, nil
+}
+
+func (proto *requestsMock) Subscribe(ctx context.Context, messages chan<- *protocol.Message, options protocol.SubscribeOptions) (*protocol.Subscription, error) {
+	return nil, nil
+}
+
+func (proto *requestsMock) Request(ctx context.Context, params protocol.RequestOptions) error {
+	proto.requests = append(proto.requests, params)
+	return nil
+}
+
+func TestRequestHistory(t *testing.T) {
```

Comment:
This is a long unit test while it seems easy:
- can we split it?
- can we at least divide it into encapsulated pieces of logic with comments describing what happens?